include/deal.II/lac/read_write_vector.h

```diff
@@ -0,0 +1,730 @@
+     *
+     * This function is analog to the the @p swap function of all C++
+     * standard containers. Also, there is a global function
+     * <tt>swap(u,v)</tt> that simply calls <tt>u.swap(v)</tt>, again in
+     * analogy to standard functions.
+     */
+    void swap (ReadWriteVector<Number> &v);
+
+    /**
+     * Copies the data and the IndexSet of the input vector @p in_vector.
+     */
+    ReadWriteVector<Number> &
+    operator= (const ReadWriteVector<Number> &in_vector);
+
+    /**
+     * Copies the data and the IndexSet of the input vector @p in_vector.
+     */
+    template <typename Number2>
+    ReadWriteVector<Number> &
+    operator= (const ReadWriteVector<Number2> &in_vector);
+
+#ifdef DEAL_II_WITH_PETSC
+    /**
+     * Imports all the elements present in the vector's IndexSet from the input
+     * vector @p petsc_vec.
+     */
+    ReadWriteVector<Number> &
+    operator= (const PETScWrappers::MPI::Vector &petsc_vec);
+#endif
+
+#ifdef DEAL_II_WITH_TRILINOS
+    /**
+     * Imports all the elements present in the vector's IndexSet from the input
+     * vector @p trilinos_vec.
+     */
+    ReadWriteVector<Number> &
+    operator= (const TrilinosWrappers::MPI::Vector &trilinos_vec);
+#endif
+
+    /**
+     * Sets all elements of the vector to the scalar @p s. This operation is
+     * only allowed if @p s is equal to zero.
+     */
+    ReadWriteVector<Number> &operator = (const Number s);
+
+    /**
+     * Returns the size of the vector.
+     *
+     * Note that the result is not equal to the number of element stored locally.
+     * The latter information is returned by n_elements()
+     */
+    size_type size() const;
+
+    /**
+     * Return the number of elements stored locally.
+     */
+    size_type n_elements() const;
+
+    /**
+     * Return the IndexSet that stores the indices of the elements stored.
+     */
+    const IndexSet &get_stored_elements () const;
+
+    /**
+     * Make the @p ReadWriteVector class a bit like the <tt>vector<></tt> class of
+     * the C++ standard library by returning iterators to the start and end
+     * of the <i>locally stored</i> elements of this vector.
+     */
+    iterator begin ();
+
+    /**
+     * Returns constant iterator to the start of the locally stored elements
+     * of the vector.
+     */
+    const_iterator begin () const;
+
+    /**
+     * Returns an iterator pointing to the element past the end of the array
+     * of locally stored entries.
+     */
+    iterator end ();
+
+    /**
+     * Returns a constant iterator pointing to the element past the end of
+     * the array of the locally stored entries.
+     */
+    const_iterator end () const;
+    //@}
+
+
+    /**
+     * @name 2: Data-Access
+     */
+    //@{
+
+    /**
+     * Read access to the data in the position corresponding to @p
+     * global_index.
+     */
+    Number operator () (const size_type global_index) const;
```

Comment:
Here and for the next four functions, can you add what happens if you access an element that is not actually stored by the current object?